Jenkinsfile.cloud

```diff
@@ -83,27 +72,33 @@ node(NODE) {
     stage("Prepare Configs") {
         withCredentials([
             string(credentialsId: params.OSTREE_INSTALL_URL, variable: 'OSTREE_INSTALL_URL'),
-            string(credentialsId: params.INSTALLER_TREE_URL, variable: 'INSTALLER_TREE_URL'),
         ]) {
-            sh "sed -i 's,\\(<url>\\).*\\(<\\/url\\),\\1${INSTALLER_TREE_URL}\\2,' rhcos.tdl"
             sh "sed -i 's,@@OSTREE_INSTALL_URL@@,${OSTREE_INSTALL_URL},' cloud.ks"
             sh "sed -i 's,@@OSTREE_INSTALL_REF@@,${ref},' cloud.ks"
-            sh "rm -rf /var/lib/imagefactory/storage/*"
         }
     }
 
-    stage("Running imagefactory") {
-        sh """
-            imagefactory --debug base_image \
-                --file-parameter install_script cloud.ks \
-                --parameter offline_icicle True \
-                rhcos.tdl
+    // Note this runs unprivileged to avoid the virt stack trying to use privileges
+    // We run in bwrap which sets up a pid namespace +init to ensure qemu is reaped,
+    // which libvirt relies on.
+    try {
+    withCredentials([
+        string(credentialsId: params.INSTALLER_TREE_URL, variable: 'INSTALLER_TREE_URL')]) {
+    stage("Running coreos-virt-install") { sh """
+        chmod a+rw /dev/kvm
+        setfacl -m u:builder:rwX ${WORKSPACE}
+        bwrap --bind / / --dev-bind /dev /dev --proc /proc --unshare-pid runuser -u builder -- coreos-virt-install --dest /tmp/rhcos.qcow2 --create-disk \
```

Comment:
For consistency, let's define `image = "/tmp/rhcos.qcow2"` in this step rather than in the next one and pass it to `coreos-virt-install`?